lxd/container_lxc.go

```diff
@@ -234,17 +234,26 @@ func lxcValidConfig(rawLxc string) error {
 		if strings.HasPrefix(key, networkKeyPrefix) {
 			fields := strings.Split(key, ".")
 
-			allowedIPKeys := []string{"ipv4.address", "ipv6.address"}
 			if !util.RuntimeLiblxcVersionAtLeast(2, 1, 0) {
-				allowedIPKeys = []string{"ipv4", "ipv6"}
-			}
+				// lxc.network.X.ipv4 or lxc.network.X.ipv6
+				if len(fields) == 4 && shared.StringInSlice(fields[3], []string{"ipv4", "ipv6"}) {
+					continue
```

Comment:
Just to clarify: we intentionally ignore invalid keys like `lxc.net.ipv4.<suffix>`?

Reply:
Nope, the check is for exactly 4 fields with the last one being ipv4 or ipv6.

Reply:
Anything that's not an exact match here hits the fallback error message.